org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/migration/AbstractRemoveDoEntityContributionValueMigrationHandlerTest.java

```diff
@@ -0,0 +1,276 @@
+   * Tests that the contribution is removed from the data object.
+   */
+  @Test
+  public void testMigrationOfDoWithMatchingAndNonExistingContribution() {
+    testMigration(createDataObjectWithUnknownContribution(), createSimpleFixtureDo());
+  }
+
+  /**
+   * Combination of multiple contributions.
+   */
+  @Test
+  public void testMigrationOfDoWithMultipleContributions() {
+    // Existing contribution 1 - should not be removed because the contribution still exists in the code
+    IDoEntity expectedExistingContribution1 = BEANS.get(DoEntity.class);
+    expectedExistingContribution1.put(DoStructureMigrationHelper.TYPE_ATTRIBUTE_NAME, "scout.FirstSimpleContributionFixture");
+    expectedExistingContribution1.put(DoStructureMigrationHelper.TYPE_VERSION_ATTRIBUTE_NAME, AlfaFixture_1.VERSION.unwrap());
+    expectedExistingContribution1.put("firstValue", "myFirstValue");
+
+    // Existing contribution 2 - no migration handler
+    IDoEntity expectedExistingContribution2 = BEANS.get(DoEntity.class);
+    expectedExistingContribution2.put(DoStructureMigrationHelper.TYPE_ATTRIBUTE_NAME, "scout.SecondSimpleContributionFixture");
+    expectedExistingContribution2.put(DoStructureMigrationHelper.TYPE_VERSION_ATTRIBUTE_NAME, AlfaFixture_1.VERSION.unwrap());
+    expectedExistingContribution2.put("secondValue", "mySecondValue");
+
+    // Unknown contribution 1 - should not be removed because of different type version
+    IDoEntity expectedUnknownContribution1 = BEANS.get(DoEntity.class);
+    expectedUnknownContribution1.put(DoStructureMigrationHelper.TYPE_ATTRIBUTE_NAME, "scout.MyTestContribution");
+    expectedUnknownContribution1.put(DoStructureMigrationHelper.TYPE_VERSION_ATTRIBUTE_NAME, AlfaFixture_2.VERSION.unwrap());
+    expectedUnknownContribution1.put("testValue", "myTestValue");
+
+    // Unknown contribution 2 - no migration handler
+    IDoEntity expectedUnknownContribution2 = BEANS.get(DoEntity.class);
+    expectedUnknownContribution2.put(DoStructureMigrationHelper.TYPE_ATTRIBUTE_NAME, "scout.MyOtherTestContribution");
+    expectedUnknownContribution2.put(DoStructureMigrationHelper.TYPE_VERSION_ATTRIBUTE_NAME, AlfaFixture_1.VERSION.unwrap());
+    expectedUnknownContribution2.put("otherTestValue", "myOtherTestValue");
+
+    // build the DO manually, because we want to add unknown/invalid contributions
+    IDoEntity expectedDataObject = BEANS.get(DoEntity.class);
+    expectedDataObject.put(DoStructureMigrationHelper.TYPE_ATTRIBUTE_NAME, "scout.SimpleFixture");
+    expectedDataObject.put("name1", "myName1");
+    expectedDataObject.putList("_contributions", CollectionUtility.arrayList(expectedExistingContribution1, expectedExistingContribution2, expectedUnknownContribution1, expectedUnknownContribution2));
+
+    // serialize and deserialize the DO, so that we get a SimpleFixtureDo instead of a DoEntity
+    IDoEntity expected = s_dataObjectMapper.readValue(s_dataObjectMapper.writeValue(expectedDataObject), IDoEntity.class);
```

Comment:
serialize / deserialize really needed? The internal list of contributions has type IDoEntity. Why does putList("_contributions") not work on a typed data object?